### ISSUE TITLE
[Bug]: instanceId missing with release 3.1.0

### ISSUE THREAD

**Issue by craigatgoogle** [jenkinsci/google-compute-engine-plugin#85]
Reported on #gcp-jenkins.
This issue appears to be breaking the plugin for customers

**Comment by thisdudeiknew** [comment]
Context: Upon upgrading from v3.0.0 to v3.1.0, I was unable to complete builds because GCE couldn't create an instance due to "missing" information in the Jenkins config.xml:
`  <clouds>
    <com.google.jenkins.plugins.computeengine.ComputeEngineCloud plugin="google-compute-engine@3.1.0">
      <name>gce-gce</name>
      <instanceCap>5</instanceCap>
      <projectId>asdf</projectId>
      <credentialsId>asdf</credentialsId>
      <instanceId></instanceId>
      <configurations>
        <com.google.jenkins.plugins.computeengine.InstanceConfiguration>
          <description>Dynamically created GCE instance for Jenkins builds</description>
          <namePrefix>jenkins-dynamic-slave</namePrefix>
          <region>https://www.googleapis.com/compute/v1/projects/asdf/regions/us-east1</region>
          <zone>https://www.googleapis.com/compute/v1/projects/asdf/zones/us-east1-b</zone>
          <machineType>https://www.googleapis.com/compute/v1/projects/asdf/zones/us-east1-b/machineTypes/n1-standard-8</machineType>
          <numExecutorsStr>1</numExecutorsStr>
          <startupScript></startupScript>
          <preemptible>false</preemptible>
          <minCpuPlatform></minCpuPlatform>
          <labels>jenkins-dynamic-slave-gcp</labels>
          <runAsUser>jenkins</runAsUser>
          <bootDiskType>https://www.googleapis.com/compute/v1/projects/asdf/zones/us-east1-b/diskTypes/pd-ssd</bootDiskType>
          <bootDiskAutoDelete>true</bootDiskAutoDelete>
          <bootDiskSourceImageName>https://www.googleapis.com/compute/v1/projects/asdf/global/images/jenkins-agent-1554304368</bootDiskSourceImageName>
          <bootDiskSourceImageProject>asdf</bootDiskSourceImageProject>
          <networkConfiguration class="com.google.jenkins.plugins.computeengine.AutofilledNetworkConfiguration">
            <network>https://www.googleapis.com/compute/v1/projects/asdf/global/networks/default</network>
            <subnetwork>default</subnetwork>
          </networkConfiguration>
          <externalAddress>true</externalAddress>
          <useInternalAddress>true</useInternalAddress>
          <networkTags></networkTags>
          <serviceAccountEmail></serviceAccountEmail>
          <mode>NORMAL</mode>
          <retentionTimeMinutesStr>11</retentionTimeMinutesStr>
          <launchTimeoutSecondsStr>300</launchTimeoutSecondsStr>
          <bootDiskSizeGbStr>50</bootDiskSizeGbStr>
          <oneShot>false</oneShot>
          <template></template>
          <windows>false</windows>
          <windowsPasswordCredentialsId></windowsPasswordCredentialsId>
          <windowsPrivateKeyCredentialsId></windowsPrivateKeyCredentialsId>
          <windowsConfig/>
          <createSnapshot>false</createSnapshot>
          <remoteFs></remoteFs>
          <googleLabels>
            <entry>
              <string>jenkins_cloud_id</string>
              <string></string>
            </entry>
            <entry>
              <string>jenkins_config_name</string>
              <string>jenkins-dynamic-slave</string>
            </entry>
          </googleLabels>
          <numExecutors>1</numExecutors>
          <retentionTimeMinutes>11</retentionTimeMinutes>
          <launchTimeoutSeconds>300</launchTimeoutSeconds>
          <bootDiskSizeGb>50</bootDiskSizeGb>
        </com.google.jenkins.plugins.computeengine.InstanceConfiguration>
      </configurations>
    </com.google.jenkins.plugins.computeengine.ComputeEngineCloud>
  </clouds>`

**Comment by thisdudeiknew** [comment]
Particularly, the `instanceId` and `jenkins_cloud_id` are missing.

**Comment by thisdudeiknew** [comment]
Error message from Jenkins:
``````30-Apr-2019 13:01:57.881 WARNING [jenkins.util.Timer [#10]] com.google.jenkins.plugins.computeengine.ComputeEngineCloud.availableNodeCapacity An error occurred counting the number of existing instances in cloud gce: 400 Bad Request
{
  "code" : 400,
  "errors" : [ {
    "domain" : "global",
    "message" : "Invalid value for field 'filter': '(labels.jenkins_cloud_id eq )'. Invalid list filter expression.",
    "reason" : "invalid"
  } ],
  "message" : "Invalid value for field 'filter': '(labels.jenkins_cloud_id eq )'. Invalid list filter expression."
}
30-Apr-2019 13:01:57.882 WARNING [jenkins.util.Timer [#10]] com.google.jenkins.plugins.computeengine.ComputeEngineCloud.provision Error provisioning node
 com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
{
  "code" : 400,
  "errors" : [ {
    "domain" : "global",
    "message" : "Invalid value for field 'filter': '(labels.jenkins_cloud_id eq )'. Invalid list filter expression.",
    "reason" : "invalid"
  } ],
  "message" : "Invalid value for field 'filter': '(labels.jenkins_cloud_id eq )'. Invalid list filter expression."
}
    at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:146)
    at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:113)
    at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:40)
    at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:321)
    at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1067)
    at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:419)
    at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
    at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
    at com.google.jenkins.plugins.computeengine.client.ComputeClient.getInstancesWithLabel(ComputeClient.java:398```

**Comment by rachely3n** [comment]
Fixed with: #86 Thanks to @ingwarsw 

I've also blacklisted the release 3.1.0 here:
https://github.com/jenkins-infra/update-center2/pull/272
